### ISSUE TITLE
Handling unlock events in Nakamoto

### ISSUE THREAD

**Issue by obycode** [stacks-network/stacks-core#5455]
When a token unlock occurs in Stacks 2.x, a mint event is attached to the coinbase transaction. This is what the API uses to track these unlocks. 

These unlocks are based on Stacks block height, not tenure height or bitcoin block height, so in 3.x, the Stacks block at which an unlock occurs might not have a coinbase transaction, since only the first block in a tenure includes a coinbase now. The stacks-node needs to attach these events somewhere else. In the current implementation, the events are just dropped if the block does not have a coinbase, causing a discrepancy in the API's view of the balances and total supply.

**Comment by hstove** [comment]
This has to be epoch-gated, right?

**Comment by obycode** [comment]
I don't think so. I think we can change events without breaking consensus. The API will handle back-filling the missed events on migration, then I think the fixed version could emit all events, including the ones that were previously missed, when booting from genesis.

**Comment by hstove** [comment]
Ok, so we're not going to actually "include" this TX in a block - we basically want to emit a "phantom" tx with these events?

**Comment by aldur** [comment]
Is this required before we can ship #5446?

**Comment by obycode** [comment]
It's a separate issue from #5446, but more urgent I'd say.
